gpcontrib/arenadata_toolkit/src/track_files.c

```diff
@@ -0,0 +1,1228 @@
+#include "postgres.h"
+
+
+#include "access/genam.h"
+#include "access/xact.h"
+#include "catalog/indexing.h"
+#include "catalog/pg_namespace.h"
+#include "catalog/pg_db_role_setting.h"
+#include "cdb/cdbdisp_query.h"
+#include "cdb/cdbdispatchresult.h"
+#include "cdb/cdbvars.h"
+#include "cdb/cdbutil.h"
+#include "commands/dbcommands.h"
+#include "executor/spi.h"
+#include "fmgr.h"
+#include "funcapi.h"
+#include "libpq-fe.h"
+#include "miscadmin.h"
+#include "storage/shmem.h"
+#include "utils/relcache.h"
+#include "utils/builtins.h"
+#include "utils/memutils.h"
+#include "utils/fmgroids.h"
+#include "utils/syscache.h"
+#include "utils/lsyscache.h"
+
+#include "arenadata_toolkit_guc.h"
+#include "arenadata_toolkit_worker.h"
+#include "drops_track.h"
+#include "dbsize.h"
+#include "file_hook.h"
+#include "tf_shmem.h"
+
+PG_FUNCTION_INFO_V1(tracking_register_db);
+PG_FUNCTION_INFO_V1(tracking_unregister_db);
+PG_FUNCTION_INFO_V1(tracking_set_snapshot_on_recovery);
+PG_FUNCTION_INFO_V1(tracking_register_schema);
+PG_FUNCTION_INFO_V1(tracking_unregister_schema);
+PG_FUNCTION_INFO_V1(tracking_set_relkinds);
+PG_FUNCTION_INFO_V1(tracking_set_relstorages);
+PG_FUNCTION_INFO_V1(tracking_is_segment_initialized);
+PG_FUNCTION_INFO_V1(tracking_trigger_initial_snapshot);
+PG_FUNCTION_INFO_V1(tracking_is_initial_snapshot_triggered);
+PG_FUNCTION_INFO_V1(tracking_get_track);
+PG_FUNCTION_INFO_V1(tracking_get_track_main);
+
+typedef struct
```

Comment:
Please add comments for each struct and field inside the struct (here and below).

Reply:
still valid